rotkehlchen/chain/evm/decoding/hop/decoder.py

```diff
@@ -333,13 +342,77 @@ def _decode_add_liquidity(self, context: DecoderContext) -> DecodingOutput:
             elif (
                 event.event_type == HistoryEventType.RECEIVE and
                 event.event_subtype == HistoryEventSubType.NONE and
-                event.location_label == user_address
+                event.location_label == user_address and
+                event.asset.is_evm_token()
             ):
                 event.event_type = HistoryEventType.RECEIVE
                 event.event_subtype = HistoryEventSubType.RECEIVE_WRAPPED
                 event.counterparty = CPT_HOP
                 event.notes = f'Receive {event.balance.amount} {event.asset.symbol_or_name()} after providing liquidity in Hop'  # noqa: E501
                 in_event = event
+                lp_token = event.asset.resolve_to_evm_token()
+                GlobalDBHandler.edit_evm_token(
+                    entry=EvmToken.initialize(
+                        address=lp_token.evm_address,
+                        chain_id=self.evm_inquirer.chain_id,
+                        token_kind=EvmTokenKind.ERC20,
+                        name=lp_token.name,
+                        symbol=lp_token.symbol,
+                        decimals=lp_token.decimals,
+                        protocol=HOP_PROTOCOL_LP,
+                    ),
+                )
+                with GlobalDBHandler().conn.write_ctx() as write_cursor:
+                    globaldb_set_unique_cache_value(
+                        write_cursor=write_cursor,
+                        key_parts=(
+                            CacheType.HOP_POOL_ADDRESS,
+                            str(lp_token.chain_id.value),
+                            lp_token.evm_address,
+                        ),
+                        value=context.tx_log.address,
+                    )
+
+                    # Query assets in the pool
+                    contract = EvmContract(
+                        address=context.tx_log.address,
+                        abi=self.evm_inquirer.contracts.abi('HOP_POOL'),
+                        deployed_block=0,  # not used here
+                    )
+                    index, asset_addresses = 0, []
+                    try:
+                        while (response := contract.call(
+                                node_inquirer=self.evm_inquirer,
+                                method_name='getToken',
+                                arguments=[index],
+                        )) is not None:
+                            index += 1
```

Comment:
do you need to keep track of the index variable? You can do len(asset_addresses)